.github/workflows/release.yml

```
@@ -84,11 +84,27 @@ jobs:
         uses: google-github-actions/setup-gcloud@v2
         with:
           version: "469.0.0"
-      - name: Upload zip
+      - name: Upload zip to GCP
         uses: google-github-actions/upload-cloud-storage@v2
         with:
           path: ${{ env.OUTNAME }}.zip
           destination: ${{ secrets.GCP_BUCKET }}/${{ github.ref_name }}/
+      - name: Copy build
+        shell: bash
+        run: |
+          mkdir build-zip
+          cp ${{ env.OUTNAME }}.zip build-zip
+      - name: Upload zip to R2
+        uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --acl public-read --follow-symlinks
+        env:
+            AWS_S3_BUCKET: ${{ secrets.CLOUDFLARE_GO_SM_BUILDS_BUCKET }}
+            AWS_ACCESS_KEY_ID: ${{ secrets.CLOUDFLARE_GO_SM_BUILDS_ACCESS_KEY_ID }}
+            AWS_SECRET_ACCESS_KEY: ${{ secrets.CLOUDFLARE_GO_SM_BUILDS_SECRET_ACCESS_KEY }}
+            SOURCE_DIR: build-zip
+            DEST_DIR: '${{ github.ref_name }}'
```

NIT: the apostrophes aren't needed 🙂 

```suggestion
            DEST_DIR: ${{ github.ref_name }}
```